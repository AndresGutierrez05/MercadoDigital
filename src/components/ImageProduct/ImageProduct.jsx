import React from "react";
import Quantity from "../Quantity/Quantity";
import "./ImageProduct.scss"

export default function ImageProduct({ src, alt, quantity }){

    return(
        <div className="container-image">
            <div className="image">
                <img src={src} alt={alt}></img>
            </div>
            
            {quantity > 0 ?
                <Quantity quantity={quantity}/>
            :
            <></>}
        </div>
    )
}