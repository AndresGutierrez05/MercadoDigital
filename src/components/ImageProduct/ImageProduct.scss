@use "sass:map";
@use "../../commonstyles.scss" as common;

.container-image{
    position: relative;
    width: 250px;
    height: 250px;

    @media screen and (max-width: common.$medium) {
        width: 150px;
        height: 150px;
        margin: 5px;
    }

    .image{
        width: 100%;
        height: 100%;
        position: absolute;
    }

    .quantity{
        width: 50px;
        height: 50px;
        position: absolute;
        color: white;
        background-color: map.get(common.$theme-colors, "purple");
        border-radius: 50%;
        box-shadow: 0 0 3px 3px map.get(common.$theme-colors, "orange");
    }
}