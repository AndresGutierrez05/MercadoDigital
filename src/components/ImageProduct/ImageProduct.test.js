import ImageProduct from "./ImageProduct.jsx";
import renderer from "react-test-renderer";

test("Image product component test", () => {
    const component = renderer.create(
        <ImageProduct />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});