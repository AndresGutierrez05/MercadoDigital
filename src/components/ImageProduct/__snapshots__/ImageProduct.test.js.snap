// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image product component test 1`] = `
<div
  className="container-image"
>
  <div
    className="image"
  >
    <img />
  </div>
</div>
`;
